src/Checker.java

```diff
@@ -0,0 +1,31 @@
+public class Checker {
```

Comment:
This is really the guts of the whole thing ay!

This is the engine, the rules. 

I think this was hiding in our original diagrams with the "Game" boxes.